GameServer.cpp

```diff
@@ -363,7 +363,7 @@ void GameServer::DoTimedUpdates()
 {
 	int i;
 	time_t currentTime;
-	unsigned int timeDiff;
+	time_t timeDiff;

```

Comment:
That's probably an improvement over what is there now.

One point of caution is that `time_t` represents a point in time, while a difference of them represents a duration of time. There might be a related data type that is more suitable for a time difference.

----

Reference: [`std::time_t`](https://en.cppreference.com/w/cpp/chrono/c/time_t)

Possibly related:
[`std::difftime`](https://en.cppreference.com/w/cpp/chrono/c/difftime)  (but uses `double`)
[`std::chrono::duration`](https://en.cppreference.com/w/cpp/chrono/duration) (looks a bit complicated)